Using UserAdmin with AbstractBaseUser

# Bug report

## What's wrong

I cant use UserAdmin for a user that's based on AbstractBaseUser and I cant import BaseUserAdmin.

```
saskatoon/member/admin.py:50: error: Type argument "AuthUser" of "UserAdmin" must be a subtype of "AbstractUser"  [type-var]
```

```
ImportError: cannot import name 'BaseUserAdmin' from 'django.contrib.auth.admin' (/app/.tox/mypy/lib/python3.9/site-packages/django/contrib/auth/admin.py)

```

The class declaration is as such: 
```
class AuthUserAdmin(UserAdmin[AuthUser]):
```

See https://github.com/LesFruitsDefendus/saskatoon-ng/blob/mypy_django/saskatoon/member/admin.py for the actual file.

## How is that should be

I dont know what I'm supposed to do.

## System information

- OS: debian bullseye
- `python` version: 3.9.23
- `django` version: 3.2.25
- `mypy` version: 1.15.0
- `django-stubs` version: 5.1.3
- `django-stubs-ext` version: 5.1.3

--- a/django-stubs/contrib/auth/admin.pyi
+++ b/django-stubs/contrib/auth/admin.pyi
@@ -1,15 +1,13 @@
-from typing import Any, TypeVar
+from typing import Any
 
 from django.contrib import admin
-from django.contrib.auth.models import AbstractUser, Group
+from django.contrib.auth.models import Group, _UserType
 from django.http.request import HttpRequest
 from django.http.response import HttpResponse
 
-_AbstractUserT = TypeVar("_AbstractUserT", bound=AbstractUser)
-
 class GroupAdmin(admin.ModelAdmin[Group]): ...
 
-class UserAdmin(admin.ModelAdmin[_AbstractUserT]):
+class UserAdmin(admin.ModelAdmin[_UserType]):
     change_user_password_template: Any
     add_fieldsets: Any
     add_form: Any

